### ISSUE TITLE
await in jupyter notebooks marked as invalid-syntax

### ISSUE THREAD

**Issue by bvolkmer** [astral-sh/ty#1823]
### Summary

`await` statements are flagged as `error[invalid-syntax]: `await` outside of an asynchronous function` even though it is allowed in Jupyter cells.

### Version

ty 0.0.1-alpha.32

**Comment by sharkdp** [comment]
FYI @ntBre

**Comment by dhruvmanila** [comment]
Related: https://github.com/astral-sh/ruff/issues/17395

**Comment by MichaReiser** [comment]
The easiest for now is probably to just skip this check based on the parsing mode

**Comment by bvolkmer** [comment]
Also, `invalid-syntax` is not a rule that can be ignored in the config:
```
warning[unknown-rule]: Unknown rule `invalid-syntax`
 --> ty.toml:2:1
  |
1 | [rules]
2 | invalid-syntax = "ignore"
  | ^^^^^^^^^^^^^^
  |
```
Edit: fixed underscore typo, still applies

**Comment by MichaReiser** [comment]
> Also, invalid-syntax is not a rule that can be ignored in the config:

This is intentional as these should, except when a bug in ty or ruff, always be fixed.

**Comment by ntBre** [comment]
Could you share an example of your code? There should already be an exception to the rule for await in the module scope/at the top level of a cell:

https://github.com/astral-sh/ruff/blob/e8540d9b085980bef22cc8e1d4c6e75531cf140c/crates/ruff_python_parser/src/semantic_errors.rs#L930-L934

In a notebook with this code:

```py
import asyncio

await asyncio.sleep(1)

if True:
    await asyncio.sleep(1)

def foo():
    await asyncio.sleep(1)
```

I only see an error for the last case, which seems correct to me.

```shell
❯ ty check crates/ruff_linter/resources/test/fixtures/pylint/await_outside_async.ipynb
error[invalid-syntax]: `await` outside of an asynchronous function
 --> crates/ruff_linter/resources/test/fixtures/pylint/await_outside_async.ipynb:cell 1:9:5
  |
8 | def foo():
9 |     await asyncio.sleep(1)  # # [await-outside-async]
  |     ^^^^^^^^^^^^^^^^^^^^^^
  |

Found 1 diagnostic
```

**Comment by samuelcolvin** [comment]
@MichaReiser you marked this as closed "as completed", but there's no link to where/how it was fixed. Do you have a link to the PR that fixed it?

**Comment by ntBre** [comment]
We needed more information to know if this was a bug or not. If you have an example of an incorrect syntax error diagnostic, we can reopen the issue!
